code/headers/mq_sensor.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+#include <mq_sensors_gas_curve_c.hpp>
+#include <array>
+#include <cmath>
+
+namespace r2d2::gas_detection {
+    /**
+     * Class mq_sensor_c provides the functionality to use all mq type sensors.
+     */
+
+    template <int AmountOfGasses>
+    class mq_sensor_c : public gas_sensor_interface_c<AmountOfGasses> {
+    private:
+        hwlib::target::pin_adc &adc_pin;
+        constexpr static int board_resistance =
+            10;     /**< The load resistance on the board in kilo ohms. */
```

Comment:
Doxygen comments of variables should go above them, not next to them. That way the value will fit on the same line as the declaration as well.
Same comment for line 17 with the comment next to `gas_concentration_resistance`

Reply:
and `ro_clean_air_factor`